src/metadata-parser.h

```diff
@@ -323,6 +330,57 @@ namespace librealsense
         };
     };
 
+    template<class S, class Attribute, typename Flag>
+    class md_attribute_parser_with_crc : public md_attribute_parser<S, Attribute, Flag>
+    {
+    public: 
+        md_attribute_parser_with_crc(Attribute S::* attribute_name, Flag flag, unsigned long long offset, attrib_modifyer mod)
+            : md_attribute_parser<S, Attribute, Flag>(attribute_name, flag, offset, mod) {}
+
+    protected:
+        bool is_attribute_valid(const S* s) const override
+        {
+            if (!md_attribute_parser<S, Attribute, Flag>::is_attribute_valid(s) || !is_crc_valid(s))
+            {
+                LOG_DEBUG("Metadata CRC mismatch");
+                return false;
+            }
+
+            // Check if the attribute's flag is set
+            auto attribute_enabled = (0 != (s->flags & static_cast<uint32_t>(this->_md_flag)));
```

Comment:
Check if needed, base class also checks it?

Reply:
Done